### PR TITLE
SOUND: FEV Update

### PR DESCRIPTION
**Pull request by Nostritius**
This PR adds several new load features for the FEV loading. Also it  moves the FEVFile class into the sound packade.

### CODE REVIEW

File: src/xoreos.cpp [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include <cstdio>
 
 #include <vector>
+#include <src/sound/fevfile.h>
```
Comment (DrMcCoy) [inline src/xoreos.cpp]: (The include format is wrong here...)

But this whole block is testing code you apparently accidentally left in :P

File: src/sound/fevfile.cpp [hunk 2]
```diff
@@ -182,9 +182,32 @@ void FEVFile::readEvent(Common::SeekableReadStream &fev) {
 	event.volumeRandomization = fev.readIEEEFloatLE();
 	event.priority = fev.readUint32LE();
 	event.maxPlaybacks = fev.readUint32LE();
-	event.mode = fev.readUint32LE();
 
-	fev.skip(12);
+	uint32 mode = fev.readUint32BE();
```
Comment (DrMcCoy) [inline src/sound/fevfile.cpp]: (There's duplicate stuff in the commit message here)

**Comment by Nostritius** [comment]
The issues should now be fixed

**Comment by DrMcCoy** [comment]
Hmm, can you rename class FEVFile to FMODEvents (I guess), to be more in line with FMODSampleBank (and the rest of the similar sound classes?). Likewise, the files should then be fmodevents.h and fmodevents.cpp.

Otherwise, this looks good :)

**Comment by Nostritius** [comment]
Hmm, it's primary function is (at the moment) still a file loader, so what about FMODEventFile?

**Comment by DrMcCoy** [comment]
Sounds good to me, yes :)

**Comment by Nostritius** [comment]
The file is now renamed

**Comment by DrMcCoy** [comment]
Merged as 70fd28a44...8bda31121, thanks! :)

**Pull request closed without merging**